### PR TITLE
Implement CurrentDateUpdater and handle date-to-string representation

### PR DESCRIPTION
**Pull request by WooKaiNing**

### CODE REVIEW

**Review by mcmc101001 — CHANGES_REQUESTED** [review]
I think we should fix merge conflicts first

**Review by mcmc101001 — APPROVED** [review]
LGTM

**Comment by WooKaiNing** [comment]
Modified input.txt to only include expense/income that have a fixed date for now, as dynamic updating of EXPECTED.TXT is required for expense/income without date input -> have implemented a CurrentDate class that will write to the EXPECTED.TXT file to update placeholder to current date, but this only works on macos and ubuntu after updating runtest.sh().

not sure if its bc i didn't modify runtest.bat correctly, thus doesnt work for windows

**Comment by WooKaiNing** [comment]
Implemented CurrentDateUpdater class to dynamically update EXPECTED.TXT file with current date (for input in INPUT.TXT where date is not specified). However, this somehow only works for Java CI / build (ubuntu-latest) (pull_request) and Java CI / build (macos-latest) (pull_request).

Hence, modified input.txt to only include expense/income input with specified date for now, will need to further refine code such that runtest.bat() works too.

**Review by mcmc101001 — APPROVED** [review]

File: src/main/java/cashleh/CurrentDateUpdater.java
Comment (danielpappa) [inline src/main/java/cashleh/CurrentDateUpdater.java:1]: Great thinking and nice implementation.

File: src/main/java/cashleh/CurrentDateUpdater.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public void updateCurrentDate() throws IOException {
+        // Read the content of the file while handling line endings
+        List<String> lines = Files.readAllLines(filePath, StandardCharsets.UTF_8);
+
+        // Get the current date in the desired format
+        LocalDate currentDate = LocalDate.now();
+        String formattedDate = currentDate.toString(); // Customize the date format as needed
+
+        // Replace all instances of the placeholder with the current date
+        List<String> updatedLines = lines.stream()
```
Comment (danielpappa) [inline src/main/java/cashleh/CurrentDateUpdater.java:31]: Great coding!

**Review by danielpappa — APPROVED** [review]

**Comment by WooKaiNing** [comment]
Fixes #45

**Pull request merged**